src/annot/pos.rs

```diff
@@ -0,0 +1,427 @@
+// Copyright 2017 Nicholas Ingolia
+// Licensed under the MIT license (http://opensource.org/licenses/MIT)
+// This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Positions on a named sequence, e.g., 683,946 on chromosome IV.
+
+use std::convert::Into;
+use std::fmt::{self, Display, Formatter};
+use std::ops::Neg;
+use std::str::FromStr;
+
+use annot::contig::Contig;
+use annot::loc::Loc;
+use annot::*;
+use strand::*;
+
+/// Position on a particular, named sequence (e.g. a chromosome).
+///
+/// Parameterized over the type of the reference sequence identifier
+/// and over the strandedness of the position.
+///
+/// The display format for a `Pos` is _chr:pos(+/-)_. A stranded
+/// position must have a _(+)_ or a _(-)_, while an unstranded
+/// position does not.
+///
+/// ```
+/// # use bio_types::annot::ParseAnnotError;
+/// # fn try_main() -> Result<(), Box<ParseAnnotError>> {
+/// use bio_types::annot::pos::Pos;
+/// use bio_types::strand::ReqStrand;
+/// let start = Pos::new("chrIV".to_owned(), 683946, ReqStrand::Reverse);
+/// let start_str = start.to_string();
+/// assert_eq!(start_str, "chrIV:683946(-)");
+/// let start_str_pos = start_str.parse()?;
+/// assert_eq!(start, start_str_pos);
+/// # Ok(())
+/// # }
+/// # fn main() { try_main().unwrap(); }
+/// ```
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+pub struct Pos<R, S> {
```

Comment:
Should there be a common trait for the different strand enums? Then, this could be constrained accordingly.

Reply:
Interesting idea. Is that a recommended pattern somewhere?